### DISCUSSION TITLE
TS issue with `createColumns` in v8

### DISCUSSION THREAD

**Discussion by colindjk** [TanStack/table#3803] [Q&A] [answered]
Given this code from the examples (removed some parts for readability):

```tsx
const table = createTable<{ Row: Person }>()

const defaultColumns = table.createColumns([
  table.createGroup({
    header: 'Name',
    footer: props => props.column.id,
    columns: [
      table.createDataColumn('firstName', {
        cell: info => info.value,
        footer: props => props.column.id,
      }),
      table.createDataColumn(row => row.lastName, {
        id: 'lastName',
        cell: info => info.value,
        header: () => <span>Last Name</span>,
        footer: props => props.column.id,
      }),
    ],
  }),
])
```

I wanted to use this without column groups, however if I try to flatten this out like so:

```tsx
```tsx
const table = createTable<{ Row: Person }>()

const defaultColumns = table.createColumns([
      table.createDataColumn('firstName', {
        cell: info => info.value,
        footer: props => props.column.id,
      }),
      table.createDataColumn(row => row.lastName, {
        id: 'lastName',
        cell: info => info.value,
        header: () => <span>Last Name</span>,
        footer: props => props.column.id,
      }),
])
```

If I do this I get a massive type error referencing a few different things, but particularly noticeable:

`types of parameters coreRowModel and coreRowModel are incompatible.`. 

I am a bit confused as to why this is happening. Luckily from a functional perspective nothings really wrong here, so I haven't been blocked, but it'd be nice to get this working instead of relying on the `as any` workaround. 

Also lmk if this should be a github issue, didn't want to jump the gun and crowd up the list of issues if it's something known / being worked on rn.

**Comment by tannerlinsley** [comment, marked as answer]
Are you on the latest version?

**Reply by colindjk** [reply]
This was on version `8.0.0-alpha.19`, running a test now to see if it happens on the latest release.

**Reply by colindjk** [reply]
It seems to have been fixed in the `8.0.0-alpha.21` release, nice!
